src/nvim/shada.c

```diff
@@ -3413,8 +3413,16 @@ static ShaDaReadResult shada_read_next_item(ShaDaReadDef *const sd_reader,
     return mru_ret;
   }
 
-  const size_t length = (size_t) length_u64;
-  entry->timestamp = (Timestamp) timestamp_u64;
+  if (length_u64 > PTRDIFF_MAX) {
+    emsgf(_(RCERR "Error while reading ShaDa file: "
+            "there is an item at position %" PRIu64 " "
+            "that is stated to be too long"),
```

Comment:
@justinmk Because it is “stated to be”, you can’t generate such long entry under normal circumstances so most likely the number is wrong for one reason or another (starting from fuzzing output like in the original issue and ending with corrupted FS).